crates/turborepo-auth/src/auth/sso.rs

```diff
@@ -41,31 +38,53 @@ pub async fn sso_login<'a, T: Client + TokenClient>(
     } = *options;
 
     let sso_team = sso_team.ok_or(Error::EmptySSOTeam)?;
+    // I created a closure that gives back a closure since the `is_valid` checks do
+    // a call to get the user, so instead of doing that multiple times we have
+    // `is_valid` give back the user email.
+    //
+    // In the future I want to make the Token have some non-skewable information and
+    // be able to get rid of this, but it works for now.
+    let valid_token_callback = |message: &str, ui: &UI| {
+        let message = message.to_string();
+        let ui = *ui;
+        move |user_email: &str| {
+            println!("{}", ui.apply(BOLD.apply_to(message)));
+            ui::print_cli_authorized(user_email, &ui);
+        }
+    };
```

Comment:
I think this might be too clever as opposed to having `is_valid` return a `Result<Option<String>>` which contains the email if the token is valid and `None` if it isn't valid.